src/lizaalert/courses/mixins.py

```diff
@@ -17,3 +21,139 @@ class TimeStampedModel(models.Model):
 
     class Meta:
         abstract = True
+
+
+def order_number_mixin(step, parent_field):
+    """Order number mixin setter."""
+
+    class SaveOrderingMixin(models.Model):
+        """Ordering mixin."""
+
+        class Meta:
+            abstract = True
+
+        order_number = models.PositiveSmallIntegerField(
+            verbose_name="порядковый номер", validators=[MinValueValidator(1)], blank=True
+        )
+
+        @property
+        def order_queryset(self):
+            """Queryset for ordering."""
+            cls = type(self)
+            filter_params = {parent_field: getattr(self, parent_field)}
+            return cls.objects.filter(**filter_params)
+
+        def set_ordering(self, queryset, order_factor):
+            """
+            Устанавливает очередность глав и уроков.
+
+            self - глава/урок
+            queryset - вызываем объект более высокого уровня
+            order_factor - номер порядка, 1000 или 10.
+            """
+            if not self.id:
+                max_order_number = queryset.aggregate(Max("order_number")).get("order_number__max")
+                self.order_number = (max_order_number or 0) + order_factor
+
+        def reset_ordering(self, queryset, order_factor):
+            """
+            Переустанавливает очередность глав и уроков.
+
+            self - глава/урок
+            queryset - вызываем объект более высокого уровня
+            order_factor - номер порядка, 1000 или 10.
+            """
+            old_order_number = type(self).objects.get(id=self.id)
+            if old_order_number != self.order_number:
+                objects = queryset.order_by("order_number")
+                for position, object in enumerate(objects):
+                    object.order_number = (position + 1) * order_factor
+                queryset.model.objects.bulk_update(objects, ["order_number"])
+
+        def _set_ordering(self, queryset, order_factor):
+            """
+            Установить/переустановить очередность глав и уроков.
+
+            self - глава/урок
+            queryset - вызываем объект более высокого уровня
+            order_factor - номер порядка, 1000 или 10.
+            """
+            if not self.id:
+                max_order_number = queryset.aggregate(Max("order_number")).get("order_number__max")
+                self.order_number = (max_order_number or 0) + order_factor
+            else:
+                # получаем старый порядковый номер
+                old_order_number = type(self).objects.get(id=self.id)
+                # округляем новый порядковый номер до шага очередности
+                if self.order_number % 10 != 0:
+                    rounded_new_order_number = (self.order_number // order_factor + 1) * order_factor
+                else:
+                    rounded_new_order_number = self.order_number
```

Comment:
нужно избавить от магического значения

```suggestion
                rounded_new_order_number = self.order_number // order_factor + self.order_number % order_factor 
```